### PR TITLE
merge tiflow/engine/pkg/errors into tiflow/pkg/errors

### LINKED ISSUE #6029: The tiflow/engine/pkg/errors should be merged into tiflow/pkg/errors
### Before asking a question, make sure you have

- [X] Searched existing Stack Overflow questions
- [X] Googled your question
- [X] Searched open and closed [GitHub issues](https://github.com/pingcap/tiflow/issues?q=is%3Aissue)
- [X] Read the [documentation](https://docs.pingcap.com/tidb/stable)

### What is your question?

The two packages should be merged to reduce duplicated code.

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Henry Lonng <henry@lonng.org>

### What problem does this PR solve?

Issue Number: close #6029

There are errors package placed in different sub-package (`tiflow/engine/pkg/errors` and `tiflow/pkg/errors`), this PR try to merge them into a single one.

### What is changed and how it works?

Merge them and fix imports.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by amyangfei — COMMENTED** [review]
I pushed some fixes directly, including

- remove `check-engine-errdoc` entry from Makefile, which is not used any more
- remove deprecated `errors.toml` and check script from engine directory
- fix make check-diff-line-width

**Review by amyangfei — COMMENTED** [review]
Besides DM error docs still locate in dm directory, what is our future plan for the error document, will they be merged into a single error toml file?

File: pkg/errors/helper.go [hunk 1]
```diff
@@ -93,3 +94,47 @@ func IsRetryableError(err error) bool {
...
+	}
+	pbErr.Message = err.Error()
+	return pbErr
+}
+
+// Wrap generates a new error based on given `*cerrors.Error`, wraps the err as
+// cause error.
+// If given `err` is nil, returns a nil error, which a the different behavior
+// against `Wrap` function in pingcap/errors.
+func Wrap(rfcError *errors.Error, err error, args ...interface{}) error {
```
Comment (amyangfei) [inline pkg/errors/helper.go:135]: `Wrap` is duplicated with `WrapError`

Comment (amyangfei) [reply pkg/errors/helper.go]: removed it in the latest commit, use `WrapError` instead.

**Comment by lonng** [comment]
> Besides DM error docs still locate in dm directory, what is our future plan for the organization of error document, will they be merged into a single error toml file?

I took a look on the DM terror package and I will merge them in another PR.

**Review by amyangfei — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**